src/Entity/EntityId.php

```diff
@@ -30,6 +55,91 @@ public function getSerialization() {
 	}
 
 	/**
+	 * Returns an array with 3 elements: the foreign repository name as the first element, the local ID as the last
+	 * element and everything that is in between as the second element.
+	 *
+	 * EntityId::joinSerialization can be used to restore the original serialization from the parts returned.
+	 *
+	 * @param string $serialization
+	 * @return string[] Array containing the serialization split into 3 parts.
+	 */
+	public static function splitSerialization( $serialization ) {
+		self::assertValidSerialization( $serialization );
+
+		$parts = explode( ':', self::normalizeIdSerialization( $serialization ) );
+		$localPart = array_pop( $parts );
+		$repoName = array_shift( $parts ) ?: '';
+		$prefixRemainder = implode( ':', $parts );
+
+		return array( $repoName, $prefixRemainder, $localPart );
+	}
+
+	/**
+	 * Builds an ID serialization from the parts returned by EntityId::splitSerialization.
+	 *
+	 * @param string[] $parts
+	 * @return string
+	 */
+	public static function joinSerialization( array $parts ) {
+		return implode( ':', array_filter( $parts ) );
+	}
+
+	/**
+	 * Returns '' for local IDs and the foreign repository name for foreign IDs. For chained IDs (e.g. foo:bar:Q42) it
+	 * will return only the first part.
+	 *
+	 * @return string
+	 */
+	public function getRepoName() {
+		$parts = self::splitSerialization( $this->serialization );
+
+		return $parts[0];
+	}
+
+	/**
+	 * Returns the serialization without the first repository prefix.
+	 *
+	 * @return string
+	 */
+	public function getLocalPart() {
+		$parts = self::splitSerialization( $this->serialization );
+
+		return self::joinSerialization( array( $parts[1], $parts[2] ) );
+	}
+
+	/**
+	 * Returns true iff EntityId::getRepoName returns a non-empty string.
+	 *
+	 * @return bool
+	 */
+	public function isForeign() {
+		// not actually using EntityId::getRepoName for performance reasons
+		return strpos( $this->serialization, ':' ) > 0;
+	}
+
+	/**
+	 * @param string $id
+	 * @return string
+	 */
+	private static function normalizeIdSerialization( $id ) {
+		return ltrim( $id, ':' );
+	}
+
+	/**
+	 * @param string $id
+	 * @return string
+	 */
+	protected static function upcaseLocalId( $id ) {
```

Comment:
I discussed with Thiemo that it would be cleaner to drop the convenience function, and just copy the relevant line of code to the child constructors that need it. I don't care terribly much either way, though.